### PR TITLE
Enhance Testing and Codebase with Label-based E2E Logs, Chart Polling, and Dependency Cleanup

### PR DESCRIPTION
**Pull request by deepakpunia-suse**
Introduced label-based logging for executing end-to-end tests.
Added wait.poll for improved chart handling.
Removed dependencies and refactored code for better maintainability and performance.

### CODE REVIEW

File: tests/e2e/installations_test.go [hunk 1]
```diff
@@ -31,16 +31,19 @@ var _ = Describe("Observability Installation Test Suite", func() {
...
-	BeforeEach(func() {
+	JustBeforeEach(func() {
 		By("Creating a client session")
 		clientWithSession, err = client.WithSession(sess)
 		Expect(err).NotTo(HaveOccurred())
 	})
 
-	It("Should install monitoring chart if not already installed", func() {
-		By("Checking if the monitoring chart is already installed")
+	It("Install monitoring chart", Label("LEVEL0", "monitoring", "Author:dpunia"), func() {
```
Comment (pankajsqe) [inline tests/e2e/installations_test.go:40]: we dont require any "author" lable in test as there wont be any usecase to run tests that are author specific.

File: tests/e2e/installations_test.go [hunk 2]
```diff
@@ -60,38 +63,37 @@ var _ = Describe("Observability Installation Test Suite", func() {
...
 		}
-
-		By("Checking if the correct registry prefix is used")
-		isUsingRegistry, err := registries.CheckAllClusterPodsForRegistryPrefix(clientWithSession, cluster.ID, registrySetting.Value)
-		Expect(err).NotTo(HaveOccurred())
-		Expect(isUsingRegistry).To(BeTrue(), "Checking if using correct registry prefix")
 	})
 
-	It("Should install Alerting chart if not already installed", func() {
+	It("Install Alerting chart", Label("LEVEL0", "alerting", "Author:dpunia"), func() {
```
Comment (pankajsqe) [inline tests/e2e/installations_test.go:88]: we dont require any "author" lable in test as there wont be any usecase to run tests that are author specific.

**Review by pankajsqe — CHANGES_REQUESTED** [review]

File: tests/e2e/installations_test.go (hunk 1 above)
Comment (deepakpunia-suse) [reply tests/e2e/installations_test.go]: Fixed

**Review by pankajsqe — APPROVED** [review]
LGTM

**Pull request merged**